core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -179,21 +182,25 @@ public ProcessingType getProcessingType() {
     // #7 Apply processor interceptors.
     interceptors.addAll(0, additionalInterceptors);
 
-
     // #8 Handle errors that occur during Processor execution. This is done outside to any scheduling to ensure errors in
-    // scheduling such as RejectedExecutionException's can be handled cleanly
-    interceptors.add((processor, next) -> stream -> from(stream).concatMap(event -> just(event)
-        .transform(next)
-        .doOnEach(signal -> currentMuleContext.set(null))
-        .onErrorResume(RejectedExecutionException.class,
-                       throwable -> Mono.from(((BaseEventContext) event.getContext())
-                           .error(resolveException((Component) processor, event, throwable)))
-                           .then(Mono.empty()))
-        .onErrorResume(MessagingException.class,
-                       throwable -> {
-                         throwable = resolveMessagingException(processor).apply(throwable);
-                         return Mono.from(((BaseEventContext) event.getContext()).error(throwable)).then(Mono.empty());
-                       })));
+    // scheduling such as RejectedExecutionException's can be handled cleanly.
+    interceptors.add((processor, next) -> stream -> from(stream)
+        .concatMap(event -> just(event)
+            .transform(next)
+            .doOnEach(signal -> currentMuleContext.set(null))
+            .onErrorResume(RejectedExecutionException.class,
+                           throwable -> Mono.from(((BaseEventContext) event.getContext())
+                               .error(resolveException((Component) processor, event, throwable)))
+                               .then(Mono.empty()))
+            // If we already got a MessagingException, avoid wrapping it again
+            .onErrorResume(MessagingException.class,
+                           throwable -> {
+                             throwable = resolveMessagingException(processor).apply(throwable);
+                             return Mono.from(((BaseEventContext) event.getContext()).error(throwable)).then(Mono.empty());
+                           })
+            .onErrorResume(throwable -> Mono
```

Comment:
Is this still needed? Having it here might help avoid hanging on unexpected exceptions thrown within chain.

Reply:
just a last check, just in case.

Reply:
k